### ISSUE TITLE
pytorch version 1.0 is not forced to install in the docker file

### ISSUE THREAD

**Issue by wangluochao902** [facebookresearch/maskrcnn-benchmark#1015]
## 🐛 Bug

In the [Dockerfile](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/docker/Dockerfile), the newest PyTorch version (1.2.0) will be installed instead of 1.0
## To Reproduce

1. docker build .

**Comment by Dorozhko-Anton** [comment]
In the Dockerfile Pytorch vision is built from the repo and it will depend on [Pytorch >= 1.1.0](https://github.com/pytorch/vision/blob/eb7a0f40ca7a7e269e893c1a8ab5845085c8b219/setup.py#L66)
`pytorch_package_name = os.getenv('TORCHVISION_PYTORCH_DEPENDENCY_NAME', 'torch >= 1.1.0')
`
But before torch vision Dockerfile will build PyTorch nightly that will be used by python
https://github.com/facebookresearch/maskrcnn-benchmark/blob/24c8c90efdb7cc51381af5ce0205b23567c3cd21/docker/Dockerfile#L36-L37

**Comment by slothkong** [comment]
@Dorozhko-Anton, looking at #1056 seems like you were already convinced that there is something fishy here. 
I built the docker image a couple of weeks ago, from the master branch of the repo and managed to make the webcam demo work (opened a [PR](https://github.com/facebookresearch/maskrcnn-benchmark/pull/983) with small improvements I made to docs as well). Funny thing is what got installed in the container shouldn't work, according to the [INSTALL.md](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/INSTALL.md):

```
root@7face3f88ebe:/maskrcnn-benchmark# conda list |  grep torch
pytorch-nightly           1.2.0.dev20190718 py3.6_cuda9.0.176_cudnn7.5.1_0    pytorch
```

```
root@7face3f88ebe:/maskrcnn-benchmark# pip freeze |  grep torch
torch==1.2.0.dev20190718
torchvision==0.3.0a0+bbd363c
```

So bad news is the [Dockerfile](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/docker/Dockerfile) is not doing what it is supposed to do. Good news is nothing broke and the docs should probably be updated when is certified that all tests pass in pytorch 1.2.0.

**Comment by Dorozhko-Anton** [comment]
@slothkong that's right, docker build worked 19 days ago because torchvision was not `v0.4.0` yet.
Nevertheless, even torchvision `v0.3.0` will try to pull the latest pytorch. And maybe depending on which version of 1.2 it will bring (nightly or simple) it will keep or replace `Pytorch` installed by conda ?
